ui/src/app/shared/components/dropdown/dropdown.tsx

```diff
@@ -19,112 +17,84 @@ export interface DropDownState {
-                </div>
-                {ReactDOM.createPortal(
-                    <div
-                        className={classNames('argo-dropdown__content', {'opened': this.state.opened, 'is-menu': this.props.isMenu})}
-                        style={{top: this.state.top, left: this.state.left}}
-                        ref={el => (this.content = el)}>
-                        {children}
-                    </div>,
-                    document.body
-                )}
-            </div>
-        );
-    }
-
-    public componentWillMount() {
-        this.subscriptions = [
-            merge(
-                dropDownOpened.pipe(filter(dropdown => dropdown !== this)),
-                fromEvent(document, 'click').pipe(
-                    filter((event: Event) => {
-                        return this.content && this.state.opened && !this.content.contains(event.target as Node) && !this.el.contains(event.target as Node);
-                    })
-                )
-            ).subscribe(() => {
-                this.close();
-            }),
-            fromEvent(document, 'scroll', {capture: true}).subscribe(() => {
-                if (this.state.opened && this.content && this.el) {
-                    this.setState(this.refreshState());
-                }
-            })
-        ];
-    }
-
-    public componentWillUnmount() {
-        (this.subscriptions || []).forEach(s => s.unsubscribe());
-        this.subscriptions = [];
-    }
-
-    public close() {
-        this.setState({opened: false});
-    }
-
-    private refreshState() {
-        const anchor = this.el.querySelector('.argo-dropdown__anchor') as HTMLElement;
-        const {top, left} = anchor.getBoundingClientRect();
-        const anchorHeight = anchor.offsetHeight + 2;
-
-        const newState = {left: this.state.left, top: this.state.top, opened: this.state.opened};
         // Set top position
-        if (this.content.offsetHeight + top + anchorHeight > window.innerHeight) {
-            newState.top = top - this.content.offsetHeight - 2;
+        if (contentEl.offsetHeight + anchorTop + anchorHeight > window.innerHeight) {
+            newState.top = anchorTop - contentEl.offsetHeight - 2;
         } else {
-            newState.top = top + anchorHeight;
+            newState.top = anchorTop + anchorHeight;
         }
 
         // Set left position
-        if (this.content.offsetWidth + left > window.innerWidth) {
-            newState.left = left - this.content.offsetWidth + anchor.offsetWidth;
+        if (contentEl.offsetWidth + anchorLeft > window.innerWidth) {
+            newState.left = anchorLeft - contentEl.offsetWidth + anchorEl.offsetWidth;
         } else {
-            newState.left = left;
+            newState.left = anchorLeft;
         }
+
         return newState;
     }
 
-    private open() {
-        if (!this.content || !this.el) {
+    function open() {
+        if (!contentRef || !anchorRef) {
```

Comment:
```suggestion
        if (!contentEl || !anchorEl) {
```

you might've had a typo here since you had the elements in the previous version. the refs themselves will always return `true`

Reply:
Oh I think I missed it, thank you 👍